runner/src/simulate.rs

```diff
@@ -95,57 +116,244 @@ impl Scenario {
+        let client = reqwest::Client::builder()
+            .connect_timeout(Duration::from_secs(3))
+            .timeout(Duration::from_secs(10))
+            .build()?;
+        let mut results = Vec::with_capacity(self.peers.len());
+        for peer in self.peers.iter() {
+            // Ideally, the peer stores the available metrics endpoints and we don't need to build them,
+            // but that's not the case right now and the ceramic/recon split makes it a bit odd, as most
+            // scenarios care about the ceramic metrics, but the recon metrics are on the IPFS port.
+            if let Some(addr) = peer.ceramic_addr() {
+                let addr = addr.parse::<reqwest::Url>()?;
+                if let Some(host) = addr.host_str() {
+                    let url = format!("http://{}:{}", host, metrics_port);
+                    let resp = client.get(&url).send().await?;
+                    if !resp.status().is_success() {
+                        warn!(?resp, "metrics request failed for peer {}", addr);
+                    } else {
+                        let body = resp.text().await?;
+                        let mut found = false;
+                        for metric in body.lines() {
+                            match metric.split(' ').collect::<Vec<&str>>().as_slice() {
+                                [name, value] if *name == metric_name => {
+                                    let val = value.parse::<u64>().map_err(|e| {warn!(metric=%name, %value, "failed to parse metric: {}", e); e}).ok();
+                                    results.push(val);
+                                    found = true;
+                                    break;
+                                }
+                                _ => {}
+                            }
+                        }
+                        if !found {
+                            warn!("Failed to find metric {} for host: {}", metric_name, addr);
+                            results.push(None);
+                        }
+                    }
+                } else {
+                    warn!("Failed to parse ceramic addr for host: {}", addr);
+                }
+            }
+        }
+        Ok(results)
+    }
+
+    /// For now, most scenarios are successful if they complete without error and only EventIdSync has a criteria.
+    /// Not a result to ensure we always proceed with cleanup, even if we fail to validate the scenario.
+    async fn validate_scenario_success(&self, metrics: &GooseMetrics) -> CommandResult {
+        if !self.manager {
+            return CommandResult::Success;
+        }
+
+        match self.scenario {
+            Scenario::IpfsRpc
+            | Scenario::CeramicSimple
+            | Scenario::CeramicWriteOnly
+            | Scenario::CeramicNewStreams
+            | Scenario::CeramicQuery
+            | Scenario::CeramicModelReuse => CommandResult::Success,
+            Scenario::EventIdSync => {
```

Comment:
This currently always fails with 0 events created on other nodes. I need to figure out if I'm subscribing incorrectly or something else is going wrong. I think I can investigate that while this is reviewed and either fix here or as a follow up. The existing scenarios work as before, and this scenario is being iterated on so having it 100% correct right away isn't required imo.

Reply:
Found at least part of the problem.. working on testing the fix

Reply:
Okay, fixed some things but it still doesn't quite work.. the metric is **always** one on the other node, and it says they are in sync. I ran scenarios creating events slowly, quickly, and it always happened.